File: urbanairship-message-center/src/main/java/com/urbanairship/messagecenter/MessageCenterActivity.java
/* Copyright Airship and Contributors */

package com.urbanairship.messagecenter;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import com.urbanairship.Autopilot;
import com.urbanairship.UALog;
import com.urbanairship.UAirship;
import com.urbanairship.activity.ThemedActivity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import kotlin.Deprecated;

/**
 * Displays the Airship Message Center using {@link MessageCenterFragment}.
 *
 * @deprecated Use {@link com.urbanairship.messagecenter.ui.MessageCenterActivity} instead.
 */
@Deprecated(message = "Replace with ui.MessageCenterActivity")
public class MessageCenterActivity extends ThemedActivity {

    @Nullable
    private MessageCenterFragment messageCenterFragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Autopilot.automaticTakeOff(getApplication());

        if (!UAirship.isTakingOff() && !UAirship.isFlying()) {
            UALog.e("MessageCenterActivity - unable to create activity, takeOff not called.");
            finish();
            return;
        }

        setDisplayHomeAsUpEnabled(true);

        if (savedInstanceState != null) {
            messageCenterFragment = (MessageCenterFragment) getSupportFragmentManager().findFragmentByTag("MESSAGE_CENTER_FRAGMENT");
        }

         if (messageCenterFragment == null) {
             messageCenterFragment = MessageCenterFragment.newInstance(MessageCenter.parseMessageId(getIntent()));
             getSupportFragmentManager()
                     .beginTransaction()
                     .add(android.R.id.content, messageCenterFragment, "MESSAGE_CENTER_FRAGMENT")
                     .commitNow();
         }

        // Apply the default message center predicate
        messageCenterFragment.setPredicate(MessageCenter.shared().getPredicate());
    }


    @Override
    protected void onNewIntent(@NonNull Intent intent) {
        super.onNewIntent(intent);
        String messageId = MessageCenter.parseMessageId(intent);
        if (messageId != null) {
            messageCenterFragment.setMessageID(messageId);
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                this.finish();
                return true;
        }
        return false;
    }

}

File: urbanairship-message-center-compose/src/main/java/com/urbanairship/messagecenter/compose/ui/MessageCenterActivity.kt
package com.urbanairship.messagecenter.compose.ui

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.urbanairship.Airship
import com.urbanairship.Autopilot
import com.urbanairship.UALog
import com.urbanairship.messagecenter.MessageCenter
import com.urbanairship.messagecenter.compose.ui.theme.MessageCenterTheme
import com.urbanairship.messagecenter.messageCenter
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update

/** Activity to display a message center */
public class MessageCenterActivity: ComponentActivity() {

    private val messageIdToOpen = MutableStateFlow<String?>(null)

    override fun onCreate(savedInstanceState: Bundle?) {
        enableEdgeToEdge()

        super.onCreate(savedInstanceState)

        Autopilot.automaticTakeOff(application)

        if (!Airship.isTakingOff && !Airship.isFlying) {
            UALog.e("MessageCenterActivity - unable to create activity, takeOff not called.")
            finish()
            return
        }

        messageIdToOpen.update { MessageCenter.parseMessageId(intent) }

        // Load the theme (or the default theme if none has been set)
        val theme = Airship.messageCenter.theme

        setContent {
            val messageId = messageIdToOpen.collectAsStateWithLifecycle()

            MessageCenterTheme(
                colors = if (isSystemInDarkTheme()) theme.darkColors else theme.lightColors,
                typography = theme.typography,
                dimens = theme.dimens,
                options = theme.options
            ) {
                MessageCenterScreen(
                    state = rememberMessageCenterState(messageId = messageId.value),
                    onNavigateUp = { finish() }
                )
            }
        }
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)

        MessageCenter.parseMessageId(intent)?.let { id ->
            messageIdToOpen.update { id }
        }
    }
}
